src/gateway_api/methods/state_methods.rs

```diff
@@ -0,0 +1,56 @@
+use crate::prelude::*;
+
+#[uniffi::export]
+impl GatewayClient {
```

Comment:
async func which *uses* endpoints, but add some custom logic "on top of them" go into the mod `gateway_api/methods`

Reply:
in this file, method using `/state` endpoint...